generators/app/templates/gulp/tasks/templates.js

```diff
@@ -1,12 +1,31 @@
 'use strict';
 
 var fs = require('fs');
+var path = require('path');
 
 var templatesTask = function (gulp, plugins, config, helpers) {
 
+  function getTemplateInputData(file) {
+    try {
+      var data = JSON.parse(
+        fs.readFileSync(config.src.dataPath + path.basename(file.path) + '.json')
+      );
+    } catch(e) {
```

Comment:
@jakub300 Wow, I've overlooked this. Great catch! 😄 
Regarding directory check... I've thought about this, but this would add unnecessary complexity IMHO. Right now the directory is created automatically (for easier discovery) and I'd also like to cover the scenario in which you already have gulp running, but decide to add some data later.